templates/single-product/product-thumbnails.php

```diff
@@ -26,7 +26,10 @@
 
 $attachment_ids = $product->get_gallery_image_ids();
 
-if ( $attachment_ids && $product->get_image_id() ) {
+if ( $attachment_ids ) {
+	if ( ! $product->get_image_id() ) {
```

Comment:
This conditional logic could also get moved into `woocommerce_show_product_thumbnails()`.

Reply:
Done in `84d6d6c`